ykbuild/build.rs

```diff
@@ -0,0 +1,51 @@
+use cmake;
+use std::env;
+use num_cpus;
+use which::which;
+
+const YKLLVM: &str = "../ykllvm/llvm";
+const PROFILE: &str = "Release";
+
+fn main() {
+    if env::var("YKB_YKLLVM_INSTALL_DIR").is_ok() {
+        println!("cargo:ykllvm={}", env::var("YKB_YKLLVM_INSTALL_DIR").unwrap());
+        return;
+    }
+
+    // Ninja builds are faster, so use this if it's available.
+    let use_ninja = which("ninja").is_ok();
+    let is_debug = env::var("PROFILE").unwrap() == "debug";
+    let nprocs = format!("-j {}", num_cpus::get());
+
+    let mut ykllvm = cmake::Config::new(YKLLVM);
+    ykllvm.profile(PROFILE)
+        .generator(if use_ninja { "Ninja" } else { "Unix Makefiles" })
+        .define("LLVM_INSTALL_UTILS", "ON")
+        .define("LLVM_ENABLE_PROJECTS", "lld;clang")
```

Comment:
`clangd` is in `clang-tools-extra`. Should we add it? Without it the LSP won't work for our C/C++ code.

The user would also have to make sure this `clangd` is in `$PATH` when starting their editor (or provide their own `clangd`).